### PR TITLE
[FLINK-6943] Improve exceptions within TypeExtractionUtils#getSingleA…

### PR DESCRIPTION
**Pull request by dawidwys**
…bstractMethod

Thanks for contributing to Apache Flink. Before you open your pull request, please take the following check list into consideration.
If your changes take all of the items into account, feel free to open your pull request. For more information and/or questions please refer to the [How To Contribute guide](http://flink.apache.org/how-to-contribute.html).
In addition to going through the list, please provide a meaningful description of your changes.

- [ ] General
  - The pull request references the related JIRA issue ("[FLINK-XXX] Jira title text")
  - The pull request addresses only one issue
  - Each commit in the PR has a meaningful commit message (including the JIRA id)

- [ ] Documentation
  - Documentation has been added for new functionality
  - Old documentation affected by the pull request has been updated
  - JavaDoc for public methods has been added

- [ ] Tests & Build
  - Functionality added by the pull request is covered by tests
  - `mvn clean verify` has been executed successfully locally or a Travis build has passed

### CODE REVIEW

File: flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractionUtils.java [hunk 1]
```diff
@@ -215,22 +215,33 @@ public static Type extractTypeArgument(Type t, int index) throws InvalidTypesExc
...
-	 * @throws InvalidTypesException if the given class does not implement
-	 * @return
+	 * @param baseClass a class that is a FunctionalInterface to retrieve a SAM from
+	 * @throws InvalidTypesException if the given class does not implement FunctionalInterface
+	 * @return single abstract method of the given class
 	 */
 	public static Method getSingleAbstractMethod(Class<?> baseClass) {
+
+		if (!baseClass.isInterface()) {
+			throw new InvalidTypesException("FunctionalInterface must be an interface");
```
Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractionUtils.java]: Shouldn't this say "Given class ... must be an interface."?

Comment (dawidwys) [inline flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractionUtils.java]: Yes and no, I would say. Given class must be a FunctionalInterface which implies it has to be an interface. If you feel. How about such a message: "Given class must be a FunctionalInterface which implies it has to be an interface."?

Comment (zentol) [inline flink-core/src/main/java/org/apache/flink/api/java/typeutils/TypeExtractionUtils.java]: My main concern is consistency with other exception messages; they all include the base class.

I would suggest "... is not a FunctionalInterface."; i don't think we need the "which implies..." bit.

**Comment by zentol** [comment]
+1.

**Comment by tedyu** [comment]
lgtm

**Comment by zentol** [comment]
merging.

**Comment by dawidwys** [comment]
@zentol Are you still gonna merge it or shall I do it?

**Comment by zentol** [comment]
I'm preparing a big batch that I'll merge on the weekend. I have to re-use the pull request to get a proper run on travis, but during the week there's just too much going on.

**Comment by dawidwys** [comment]
Great, thanks!

**Pull request closed without merging**